### PR TITLE
LPD-30311 | master

### PR DESCRIPTION
**Pull request by marcosapmf**
Jira ticket: https://liferay.atlassian.net/browse/LPD-30311

### CODE REVIEW

**Comment by marcosapmf** [comment]
ci:test:sf

**Comment by marcosapmf** [comment]
ci:test:sf

**Comment by marcosapmf** [comment]
ci:test:analytics:cloud

**Comment by marcosapmf** [comment]
ci:test:analytics-cloud

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/analyticscloud/ABTestDXPSide.testcase [hunk 1]
```diff
@@ -18,8 +18,6 @@ definition {
 		}
 
 		task ("Set up instance and sign in DXP") {
-			TestCase.setUpPortalInstance();
-
 			ACUtils.launchDXP();
 		}
```
Comment (ViniLopes20) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/analyticscloud/ABTestDXPSide.testcase:22]: Hey @marcosapmf, you need to revert this

File: portal-web/test/functional/com/liferay/portalweb/macros/ABTest.macro [hunk 2]
```diff
@@ -248,7 +250,7 @@ definition {
 			key_text = "Publish",
 			locator1 = "Button#ANY");
 
-		Alert.viewSuccessMessageText(successMessage = "Success:${variantName} was published successfully.");
+		AssertVisible(locator1 = "ABTest#IMAGE_EMPTY_STATE");
```
Comment (ViniLopes20) [inline portal-web/test/functional/com/liferay/portalweb/macros/ABTest.macro:253]: Hey @marcosapmf, I think it is not correct. After publishing a variant, then the success alert should appear, why do you change to check if the empty state will appear?

**Review by ViniLopes20 — CHANGES_REQUESTED** [review]
Hey @marcosapmf, thanks for working on it. The Poshi report shows that the error is related to the `PageEditor.editFragmentText`, I don't see any change associated with it and your test result is failing.

Follow this when you try to execute your cases again: https://liferay.atlassian.net/wiki/spaces/ENGAC/pages/2895183873/How+to+run+the+Playwright+or+and+Poshi+tests+in+the+CI#%3Aci%3A-%3Aposhi%3A-How-to-run-a-specific-Poshi-test-in-the-CI%3F

**Comment by marcosapmf** [comment]
ci:report:29928657

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/analyticscloud/ABTestDXPSide.testcase (hunk 1 above)
Comment (marcosapmf) [reply portal-web/test/functional/com/liferay/portalweb/tests/enduser/analyticscloud/ABTestDXPSide.testcase:22]: ops :P

File: portal-web/test/functional/com/liferay/portalweb/macros/ABTest.macro (hunk 2 above)
Comment (marcosapmf) [reply portal-web/test/functional/com/liferay/portalweb/macros/ABTest.macro:253]: @ViniLopes20, the success message does appear. However, the page is reloaded before Poshi could check the alert. I talked to @interaminense about this, and the best option was finding another way to assert the variant was published. One way is guaranteeing that we have the empty state in AB test side panel :)

**Comment by marcosapmf** [comment]
ci:test:analytics-cloud

File: portal-web/test/functional/com/liferay/portalweb/macros/ABTest.macro [hunk 3]
```diff
@@ -126,20 +126,22 @@ definition {
...
 			}
-		}
 
-		Click(locator1 = "ACSettings#CONFIRMATION_BUTTON");
+			Click(locator1 = "ACSettings#CONFIRMATION_BUTTON");
 
-		Alert.viewSuccessMessageText(successMessage = "Success:Your request completed successfully.");
+			Alert.viewSuccessMessageText(successMessage = "Success:Your request completed successfully.");
+		}
 	}
```
[the remark this answers is not in the document]
Comment (ViniLopes20) [reply portal-web/test/functional/com/liferay/portalweb/macros/ABTest.macro:145]: and related with the `PageEditor.editFragmentText`?

File: portal-web/test/functional/com/liferay/portalweb/macros/ABTest.macro (hunk 2 above)
Comment (ViniLopes20) [reply portal-web/test/functional/com/liferay/portalweb/macros/ABTest.macro:253]: Hey @marcosapmf, the purpose of this `Alert.viewSuccessMessageText` is not to guarantee that the variant was published (we do this by checking the version that is appearing on the page). This alert is to ensure that the alert is displaying the feedback of the page to the user, if this is not showing then this is a bug.

This change should not be made

File: portal-web/test/functional/com/liferay/portalweb/macros/ABTest.macro (hunk 3 above)
[the remark this answers is not in the document]
Comment (marcosapmf) [reply portal-web/test/functional/com/liferay/portalweb/macros/ABTest.macro:145]: Got it. I'll move this logic into tearDown then. Thanks @ViniLopes20

File: portal-web/test/functional/com/liferay/portalweb/macros/ABTest.macro (hunk 2 above)
Comment (interaminense) [reply portal-web/test/functional/com/liferay/portalweb/macros/ABTest.macro:253]: Vini, this is a usability issue, currently, after the user publishes, the page has to be redirected to the published variant, so the alert no longer makes sense in this case, I would even talk to Deborah about removing this alert.

Comment (marcosapmf) [reply portal-web/test/functional/com/liferay/portalweb/macros/ABTest.macro:253]: Hey @ViniLopes20, in light of what you and Adriano said, we can just remove this alert assertion for now, right?

**Comment by marcosapmf** [comment]
ci:test:analytics-cloud

**Comment by marcosapmf** [comment]
ci:test:analytics-cloud

File: portal-web/test/functional/com/liferay/portalweb/macros/ABTest.macro (hunk 2 above)
Comment (ViniLopes20) [reply portal-web/test/functional/com/liferay/portalweb/macros/ABTest.macro]: Hey @marcosapmf, Let's see this on your machine, I'm validating it here on mine and I don't see the page being loaded again and removing the success alert.

File: portal-web/test/functional/com/liferay/portalweb/macros/layout/pageeditor/PageEditor.macro [hunk 4]
```diff
@@ -1357,6 +1357,8 @@ definition {
 			var index = 1;
 		}
 
+		WaitForVisible(locator1 = "Fragment#FRAGMENT_LABEL");
+
 		Click.javaScriptClick(
```
[the remark this answers is not in the document]
Comment (marcosapmf) [reply portal-web/test/functional/com/liferay/portalweb/macros/layout/pageeditor/PageEditor.macro:1362]: For sure, my friend :smile:

[the remark this answers is not in the document]
Comment (marcosapmf) [reply portal-web/test/functional/com/liferay/portalweb/macros/layout/pageeditor/PageEditor.macro:1362]: Hey @ViniLopes20, I just ran the following tests and they passed:
```
FragmentsConfiguration#ViewValidationInTextFieldWithEmailConstraintRules
TextInput#CannotHideTextInputFragmentOnContentPage
ContentPages#HeaderOnlyShownWhenElementSelected
```

**Comment by marcosapmf** [comment]
ci:test:sf

**Comment by marcosapmf** [comment]
ci:report:30817479

File: portal-web/test/functional/com/liferay/portalweb/macros/layout/pageeditor/PageEditor.macro (hunk 4 above)
[the remark this answers is not in the document]
Comment (ViniLopes20) [reply portal-web/test/functional/com/liferay/portalweb/macros/layout/pageeditor/PageEditor.macro:1362]: Nice @marcosapmf 🎉

**Comment by marcosapmf** [comment]
ci:test:analytics-cloud

**Comment by marcosapmf** [comment]
ci:report:31413595

**Comment by marcosapmf** [comment]
ci:test:analytics-cloud

**Comment by marcosapmf** [comment]
ci:test:sf

**Comment by marcosapmf** [comment]
ci:test:relevant

**Comment by ViniLopes20** [comment]
ci:test:stable

**Review by ViniLopes20 — APPROVED** [review]
Very good, @marcosapmf . Thank you for working on this 🎊

**Comment by ViniLopes20** [comment]
ci:test:stable

**Comment by marcosapmf** [comment]
ci:report:33140147

**Comment by marcosapmf** [comment]
ci:test:stable

**Comment by marcosapmf** [comment]
ci:report:33140147

**Comment by marcosapmf** [comment]
ci:test:relevant

**Comment by marcosapmf** [comment]
ci:test:sf

**Comment by marcosapmf** [comment]
ci:forward

**Pull request closed without merging**